lib/valkyrie/storage/fedora.rb

```diff
@@ -2,15 +2,26 @@
 module Valkyrie::Storage
   # Implements the DataMapper Pattern to store binary data in fedora
   class Fedora
-    attr_reader :connection, :base_path, :fedora_version
+    attr_reader :connection, :base_path, :fedora_version, :pairtree_count, :pairtree_length, :uri_transformer
     PROTOCOL = 'fedora://'
     SLASH = '/'
 
     # @param [Ldp::Client] connection
-    def initialize(connection:, base_path: "/", fedora_version: Valkyrie::Persistence::Fedora::DEFAULT_FEDORA_VERSION)
+    # @param [Integer] fedora_pairtree_count
+    # @param [Integer] fedora_pairtree_length
+    def initialize(connection:, base_path: "/", fedora_version: Valkyrie::Persistence::Fedora::DEFAULT_FEDORA_VERSION,
+                   fedora_pairtree_count: 0, fedora_pairtree_length: 0)
       @connection = connection
       @base_path = base_path
       @fedora_version = fedora_version
+      @pairtree_count = fedora_pairtree_count
+      @pairtree_length = fedora_pairtree_length
+
+      @uri_transformer = if fedora_version == 6 && (pairtree_count * pairtree_length).positive?
```

Comment:
I see that the storage adapter isn't doing any pair trees for Fedora 4.  I'm curious why the metadata adapter does it but the storage adapter doesn't.  This isn't really in scope of your PR though so you don't need to answer this but I wanted to raise the question anyways.